distribution/std-lib/Table/src/Data/Table.enso

```
@@ -165,6 +167,107 @@ type Table
+
+       > Example
+         Sorting `table` in ascending order by the value in column `'Quantity'`,
+         using the value in column `'Rating'` in descending order for breaking
+         ties.
+             table.sort by=['Quantity', Order_Rule 'Rating' (order=Sort_Order.Descending)]
+
+       > Example
+         Sorting `table` in ascending order by the value in an externally
+         computed column, using the value in column `'Rating'` for breaking
+         ties.
+             quality_ratio = table.at 'Rating' / table.at 'Price'
+             table.sort by=[quality_ratio, 'Rating']
+
+       > Sorting `table` in ascending order, by the value in column
+         `'position'`, using a custom comparator function.
+             manhattan_comparator a b = (a.x.abs + a.y.abs) . compare_to (b.x.abs + b.y.abs)
+             table.sort by=(Order_Rule 'position' comparator=manhattan_comparator)
+    sort : Text | Column.Column | Order_Rule | Vector.Vector (Text | Column.Column | Order_Rule) -> Sort_Order -> Boolean -> Table
+    sort by order=Sort_Order.Ascending missing_last=True =
+        rules = this.build_java_order_rules by order missing_last
+        fallback_cmp = here.comparator_to_java .compare_to
+        mask = OrderBuilder.buildOrderMask rules.to_array fallback_cmp
+        new_table = this.java_table.applyMask mask
+        Table new_table
+
+    ## PRIVATE
+    build_java_order_rules rules order missing_last = case rules of
+        Text -> [this.build_java_order_rule rules order missing_last]
+        Column.Column _ -> [this.build_java_order_rule rules order missing_last]
+        Order_Rule _ _ _ _ -> [this.build_java_order_rule rules order missing_last]
+        Vector.Vector _ -> rules.map (this.build_java_order_rule _ order missing_last)
+
+    ## PRIVATE
+    build_java_order_rule rule order missing_last =
+        order_bool = case order of
+            Sort_Order.Ascending -> True
+            _ -> False
```

Isn't it better to just add `Sort_Order.Descending` here?

If for some weird reason we get something else here, won't it be more meaningful to fail with inexhaustive pattern match saying that the argument was unexpected instead of silently falling back to some default?